src/core/qgsmapclippingregion.h

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+  qgsmapclippingregion.h
+  --------------------------------------
+  Date                 : June 2020
+  Copyright            : (C) 2020 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPCLIPPINGREGION_H
+#define QGSMAPCLIPPINGREGION_H
+
+#include "qgis_core.h"
+#include "qgsgeometry.h"
+#include "qgsmaplayer.h"
+
+/**
+ * \class QgsMapClippingRegion
+ * \ingroup core
+ *
+ * A map clipping region (in map coordinates and CRS).
+ *
+ * \since QGIS 3.16
+*/
+class CORE_EXPORT QgsMapClippingRegion
+{
+  public:
+
+    /**
+     * Feature clipping behavior.
+     */
+    enum class FeatureClippingType : int
+    {
+      Intersect, //!< Clip the geometry of these features to the region prior to rendering (i.e. feature boundaries will follow the clip region)
```

Comment:
Wondering if there's a better name for it (to try and avoid the one-letter Intersect vs Intersects difference).

Maybe OverlayClip or IntersectClip?

Reply:
... or:
- IntersectClip
- IntersectFilter

Reply:
Agreed, thanks - I've renamed all these enum values for clarity